app/views/common/os_selection/_pxe_loader.html.erb

```diff
@@ -0,0 +1,12 @@
+<%= fields_for item do |f| -%>
+  <% loaders = @operatingsystem.available_loaders(item.architecture); if loaders.any? -%>
+    <% selected_attr = f.object.new_record? && item.hostgroup.present? && item.hostgroup.pxe_loader.nil? ? { :selected => @operatingsystem.preferred_loader(item.architecture) } : {} -%>
```

Comment:
@lzap: Here is your reproducer: This does not work when you don't use a hostgroup. I guess you need to change the code to something like this:

```ruby
f.object.new_record? && (!item.hostgroup.present? || (item.hostgroup.present? && item.hostgroup.pxe_loader.nil?))
```

Then the sane default is actually suggested. And then it always jumps to the sane default when you save the host form with validation errors.